x-pack/legacy/plugins/security/public/lib/kibana_privilege_calculator/kibana_feature_privilege_calculator.ts

```diff
@@ -141,13 +150,21 @@ export class KibanaFeaturePrivilegeCalculator {
     }
 
     if (!ignoreAssigned) {
+      const actions = this.getFeatureActions(
+        featureId,
+        this.getAssignedFeaturePrivilege(privilegeSpec, featureId)
+      );
+      const directlyAssignedPrivilegeMorePermissiveThanInherited = !areActionsFullyCovered(
```

Comment:
We aren't considering global feature privileges as part of this calculation. If a space has `discover:all` set, and then the global entry also has `discover:all` set, then this returns
```js
{
  actualPrivilege: "all"
  actualPrivilegeSource: 10
  directlyAssignedPrivilegeMorePermissiveThanInherited: true
  isDirectlyAssigned: true
}
```
It's not technically true that the directly assigned privilege is more permissive than the inherited. In this case, the are equivalent to each other.

The way you have it written now solves your problem, and we likely don't care about this particular scenario. If we were going to keep this calculator around for the foreseeable future, then I'd push for a more correct fix, but it's likely not worth the effort. Maybe we solve this with a naming change?

What do you think about `directlyAssignedPrivilegeMorePermissiveThanInherited` => `directlyAssignedPrivilegeMorePermissiveThanBase`? (I don't love it).

Since this flag is only applicable within the context of feature privileges, this might make enough sense. It's already defined as an [optional boolean](https://github.com/kobelb/kibana/blob/a65f9ab59d422b904b206f0783a9cda76e9f681a/x-pack/legacy/plugins/security/public/lib/kibana_privilege_calculator/kibana_privilege_calculator_types.ts#L30), so this just won't be set for scenarios where it doesn't make sense. Adding a comment to this effect to the type definition would probably help future-me

Reply:
That's a good point, the new naming makes more sense.